tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -1351,6 +1351,27 @@ static LogicalResult Verify(BroadcastOp op) {
   return success();
 }
 
+OpFoldResult BroadcastOp::fold(ArrayRef<Attribute> attrs) {
+  auto type = getType().cast<RankedTensorType>();  
+  if (type == getOperand().getType()) {
+    auto sizesSize = broadcast_sizes().getType().getNumElements();
+    if (sizesSize != 0) {
```

Comment:
It's not clear to me what you're checking for here? Which test does this correspond to?

Reply:
I think `if (type == getOperand().getType())` is enough. I will remove this checking.